src/dune_rpc_impl/long_poll.ml

```diff
@@ -98,13 +97,28 @@ module Build_system = Dune_engine.Build_system
 module Progress = struct
   module Progress = Dune_rpc.Progress
 
-  type state = unit
+  (* If [Some], this particular poller hasn't seen this update yet. We need to
+     track this to prevent repeatedly sending the same event. *)
+  type state = Progress.t option
```

Comment:
I'd use a custom variant here to self document. E.g
```ocaml
type state =
  | Next_update of Progress.t
  | No_pending
```

Or something like that.